src/stores/torrent-store.js

```diff
@@ -57,85 +57,60 @@ class TorrentStore {
     return this.rpc.sendRequest('torrent-get', data).then(action((response) => {
       response.json().then(action((result) => {
         this.previousTorrents = this.torrents.slice();
-        this.torrents.replace(
-          result.arguments.torrents.map((torrent) => new Torrent(torrent))
-        );
-      }));
-    }));
-  }
-
-  @action fetchIds(torrentIds) {
-    const data = {
-      fields: ['id']
-    };
-
-    if (torrentIds) {
-      data['ids'] = torrentIds;
-    }
-
-    return this.rpc.sendRequest('torrent-get', data).then(action((response) => {
-      response.json().then(action((result) => {
-        this.torrents.replace(
-          result.arguments.torrents.map((torrent) => new Torrent(torrent))
-        );
+        const newTorrents = result.arguments.torrents;
+
+        if (torrentIds) {
+          // There are ids passed to fetch, edit cache of dirty torrents
+          // XXX: This is a potentially slow operation. Perhaps cache should be a map?
+          newTorrents.forEach(torrent => {
+            const i = this.torrents.findIndex(t => t.id === torrent.id);
+            if (i >= 0)
+              this.torrents[i] = new Torrent(torrent);
```

Comment:
Maybe would be interesting to implement a `set` method in `Torrent` class. That way we could reuse the same instance and we wouldn't need to recreate models again and again. What do you think @Secretmapper?